mongomock/filtering.py

```diff
@@ -486,10 +506,14 @@ def operator_eq(doc_val, search_val):
     'javascript': None,
     'symbol': None,
     'javascriptWithScope': None,
-    'int': (int,),
+    'int': lambda v: (
+        isinstance(v, int) and not isinstance(v, bool) and v.bit_length() <= 32
+    ),
     'timestamp': None,
-    'long': (float,),
-    'decimal': (float,),
+    'long': lambda v: (
+        isinstance(v, int) and not isinstance(v, bool) and v.bit_length() > 32
+    ),
+    'decimal': Decimal128,
```

Comment:
Please put Decimal128 in a tuple with one value so that this map only has values with `None` or a tuple.

If you want to cover for the case where Decimal128 is not defined, please solve it here directly `(Decimal128,) if Decimal128 else None`

Reply:
Nice. It makes it explicit. Thanks.

Reply:
Done, I'll update the PR. (Note however that this change is actually part of #755 )